core/metadata.js

```diff
@@ -217,37 +218,49 @@ function isUpToDate (side /*: SideName */, doc /*: Metadata */) {
 // Return true if the metadata of the two folders are the same
 // For updated_at, we accept up to 3s of differences because we can't
 // rely on file systems to be precise to the millisecond.
+
+const makeDiffFilter = (interestingFields) => (path, key) => {
+  return path.length === 0 && interestingFields.indexOf(key) === -1
+}
+
+const sameFolderFilter = makeDiffFilter([
+  '_id', 'docType', 'remote', 'tags', 'trashed', 'ino'])
+
 function sameFolder (one /*: Metadata */, two /*: Metadata */) {
   const {path} = two
-  let fields = ['_id', 'docType', 'remote', 'tags', 'trashed', 'ino']
-  one = pick(one, fields)
-  two = pick(two, fields)
-  const same = isEqual(one, two)
-  if (!same) log.trace({path, diff: {one, two}})
-  return same
+  let diff = delta(one, two, sameFolderFilter)
+  if (!diff || diff.length === 0) return true
```

Comment:
Quoting the doc:

> Returns either an array of changes or, if there are no changes, undefined. This was originally chosen so the result would be pass a truthy test [...]

Is it broken?

I wouldn't mind logging a `null` diff, so we don't end up wondering whether there is no change or we missed the log entry:

```js
log.trace({path, diff})
return !diff
```